File: Factory/FactoryMethod/Java/src/Pizza/NewYorkStyle/NewYorkStyleVeggiePizza.java
package eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class NewYorkStyleVeggiePizza extends Pizza {
    public NewYorkStyleVeggiePizza(){
        name = "New York Style Veggie Pizza";
        dough = "Thin Crust Dough";
        sauce = "Marinara Sauce";

        toppings.add("Grated Reggiano Cheese");
        toppings.add("Garlic");
        toppings.add("Onion");
        toppings.add("Mushrooms");
        toppings.add("Red Pepper");
    }
}

File: Factory/FactoryMethod/TypeScript/src/Pizza/NewYorkStyle/NewYorkStyleVeggiePizza.ts
import Pizza from '../Pizza';

class NewYorkStyleVeggiePizza extends Pizza {
  constructor() {
    super();
    this.name = 'New York Style Veggie Pizza';
    this.dough = 'Thin Crust Dough';
    this.sauce = 'Marinara Sauce';

    this.toppings.push('Grated Reggiano Cheese');
    this.toppings.push('Garlic');
    this.toppings.push('Onion');
    this.toppings.push('Mushrooms');
    this.toppings.push('Red Pepper');
  }
}

export default NewYorkStyleVeggiePizza;
